src/prognostic_equations/implicit/implicit_solver.jl

```diff
@@ -1091,6 +1102,40 @@ function update_implicit_equation_jacobian!(A, Y, p, dtγ)
                 @. ∂ᶜρq_tot_err_∂ᶜq_totʲ =
                     -(dtγ * ᶜadvdivᵥ_matrix() ⋅ ∂ᶜupdraft_mass_flux_∂ᶜqtot)
 
+                # ∂/∂u₃ʲ contributions to grid-mean ρe_tot and ρq_tot from updraft mass flux
+                ∂ᶜρe_tot_err_∂ᶠu₃ʲ =
+                    matrix[@name(c.ρe_tot), @name(f.sgsʲs.:(1).u₃)]
+                @. ∂ᶜρe_tot_err_∂ᶠu₃ʲ =
+                    dtγ * ᶜadvdivᵥ_matrix() ⋅
+                    DiagonalMatrixRow((ᶠinterp(ᶜρʲs.:(1) * ᶜJ) / ᶠJ),) ⋅
+                    DiagonalMatrixRow(ᶠinterp(Y.c.sgsʲs.:(1).ρa / ᶜρʲs.:(1))) ⋅
+                    DiagonalMatrixRow(g³³(ᶠgⁱʲ)) ⋅ ᶠinterp_matrix() ⋅
+                    DiagonalMatrixRow((Y.c.sgsʲs.:(1).mse + ᶜKʲs.:(1) - ᶜh_tot))
+
+                ∂ᶜρq_tot_err_∂ᶠu₃ʲ =
+                    matrix[@name(c.ρq_tot), @name(f.sgsʲs.:(1).u₃)]
+                @. ∂ᶜρq_tot_err_∂ᶠu₃ʲ =
+                    dtγ * ᶜadvdivᵥ_matrix() ⋅
+                    DiagonalMatrixRow((ᶠinterp(ᶜρʲs.:(1) * ᶜJ) / ᶠJ),) ⋅
+                    DiagonalMatrixRow(ᶠinterp(Y.c.sgsʲs.:(1).ρa / ᶜρʲs.:(1))) ⋅
+                    DiagonalMatrixRow(g³³(ᶠgⁱʲ)) ⋅ ᶠinterp_matrix() ⋅
+                    DiagonalMatrixRow(Y.c.sgsʲs.:(1).q_tot - ᶜspecific.q_tot)
+
+                # ∂/∂(rho*a) contributions to grid-mean ρe_tot and ρq_tot from updraft mass flux
+                ∂ᶜρe_tot_err_∂ᶜρa =
+                    matrix[@name(c.ρe_tot), @name(c.sgsʲs.:(1).ρa)]
+                @. ∂ᶜρe_tot_err_∂ᶜρa =
+                    dtγ * ᶜadvdivᵥ_matrix() ⋅
+                    DiagonalMatrixRow(ᶠu³ʲs.:(1) - ᶠu³) ⋅ ᶠinterp_matrix() ⋅
+                    DiagonalMatrixRow((Y.c.sgsʲs.:(1).mse + ᶜKʲs.:(1) - ᶜh_tot))
+
+                ∂ᶜρq_tot_err_∂ᶜρa =
+                    matrix[@name(c.ρq_tot), @name(c.sgsʲs.:(1).ρa)]
+                @. ∂ᶜρq_tot_err_∂ᶜρa =
+                    dtγ * ᶜadvdivᵥ_matrix() ⋅
+                    DiagonalMatrixRow(ᶠu³ʲs.:(1) - ᶠu³) ⋅ ᶠinterp_matrix() ⋅
+                    DiagonalMatrixRow((Y.c.sgsʲs.:(1).q_tot - ᶜspecific.q_tot))
```

Comment:
I think there should be a minus sign for these two, but I can double check.

Reply:
I double-checked and agree - updated.